app/Console/Commands/KeyGenerateCommand.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Encryption\Encrypter;
+use Safe\Exceptions\UrlException;
+
+class KeyGenerateCommand extends \Illuminate\Foundation\Console\KeyGenerateCommand
```

Comment:
Please add a PhpDoc comment to the class which explains why it is necessary to override the command and provide reference to the Composer install script. Other developer in the future may ask themselves the same question.

Reply:
That is not exactly what I meant. The comment only explains what the original command does anyway. This does not explain _why_ we need to overwrite it. I meant background information, not the obvious. More something like this

```php
/**
 * Generates the `APP_KEY` config variable.
 *
 * This class extends the original command by the additional option `--no-override`
 * which - if enabled - let the command silently do nothing, if the variable has already been set.
 * This special mode is needed in the automatic install script which is called by Composer
 * upon package installation
 */
```

This is what I meant by `provide reference to the Composer install script`. In 6 months from now on, nobody will remember we we needed that.